Completion/Unix/Command/_email-notmuch

```diff
@@ -0,0 +1,9 @@
+#autoload
```

Comment:
Should be in a different directory: Types/ rather than Command/.

Reply:
I can move it, but where in the zsh docs does it explain what these different directories mean?

I was only able to find this [0]

http://zsh.sourceforge.net/Doc/Release/Completion-System.html#Completion-Directories

Reply:
> where in the zsh docs does it explain what these different directories mean?

Nowhere :(

Reply:
:( indeed

Reply:
I'd add it to the link you posted, but I don't think the user manual is the right place to document the source code layout in.  I'm not sure where the information would fit better, either.  Where would you expect that bit of info to be documented?

Reply:
Looking at the user documentation I was trying to answer the question:

If a package manager is used to build and install packages (thus not locally in `/usr/local`), where should they put the zsh completions? I'm assuming that /usr/local/zsh/functions is probably for internal completions for zsh itself, but packages seem to install their completions into those subdirectories anyways - at least that's the case on my system now that I've been looking at it.

Do packages install their zsh completions into the /usr/share/zsh/{functions,site-functions,vendor-completions,???}.

That isn't stated in the docs, but might be useful for package managers

Reply:
> packages seem to install their completions into those subdirectories anyways - at least that's the case on my system now that I've been looking at it.

This is controlled by the `--enable-function-subdirs` configure flag.

> Do packages install their zsh completions into the /usr/share/zsh/{functions,site-functions,vendor-completions,???}.

`site-functions` is reserved to the local sysadmin. `vendor-completions` is probably added to fpath by your distro as a place for distro packages to put their scripts in (that's how Debian does it, anyway). `$PREFIX/share/zsh/$ZSH_VERSION/functions` is where `make install` places the functions shipped with zsh itself.  Does your use-case fall into one of these?

This is a recurring question — the debian zsh packagers get it too from time to time — so we should probably add the answer to the docs.

Reply:
`vendor-completions` is purely a Debian/Ubuntu thing.

Normally packages install their functions into `$prefix/share/zsh/site-functions`. That is at least going to be in the default $fpath for zsh users. It is probably up to distributions to separate things out.